src/Relations/MorphToMany.php

```diff
@@ -0,0 +1,407 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Laravel\Relations;
+
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\MorphToMany as EloquentMorphToMany;
+use Illuminate\Support\Arr;
+
+use function array_diff;
+use function array_key_exists;
+use function array_keys;
+use function array_map;
+use function array_merge;
+use function array_reduce;
+use function array_values;
+use function count;
+use function is_array;
+use function is_numeric;
+
+class MorphToMany extends EloquentMorphToMany
+{
+    /**
+     * Get the key for comparing against the parent key in "has" query.
+     *
+     * @return string
+     */
+    public function getHasCompareKey()
+    {
+        return $this->foreignPivotKey;
+    }
```

Comment:
I guess this won't work. I will write some tests for that tomorrow.

Reply:
It's more complicated than I thought. I think we have to make some changes [here](https://github.com/mongodb/laravel-mongodb/blob/1c2d1fe29a53278ec34a50c1a289b5e3a8f1a838/src/Helpers/QueriesRelationships.php#L117) to fix this issue.

Reply:
I removed this method because it has no use in `has` querying.